src/common/ceph_context.cc

```diff
@@ -386,9 +386,13 @@ class CephContextObs : public md_config_obs_t {
     if (changed.count(
 	  "enable_experimental_unrecoverable_data_corrupting_features")) {
       std::lock_guard lg(cct->_feature_lock);
-      get_str_set(
-	conf->enable_experimental_unrecoverable_data_corrupting_features,
-	cct->_experimental_features);
+
+      auto func = [_cct = cct] (auto feature) {
```

Comment:
```suggestion
      auto add_experimental_feature = [cct] (auto feature) {
```

Reply:
@tchaikov 
class member can't be captured. Refer to: https://en.cppreference.com/w/cpp/language/lambda
```
Class members cannot be captured explicitly by a capture without initializer
```
To align with your idea, I'll capture "this" pointer here.